pygmt/clib/session.py

```diff
@@ -494,6 +494,97 @@ def get_default(self, name):
 
         return value.value.decode()
 
+    def get_common(self, option):
+        """
+        Inquire if a GMT common option has been set and return its current
+        value if possible.
+
+        Parameters
+        ----------
+        option : str
+            The GMT common option to check. Valid options are ``"B"``, ``"I"``,
+            ``"J"``, ``"R"``, ``"U"``, ``"V"``, ``"X"``, ``"Y"``, ``"a"``,
+            ``"b"``, ``"f"``, ``"g"``, ``"h"``, ``"i"``, ``"n"``, ``"o"``,
+            ``"p"``, ``"r"``, ``"s"``, ``"t"``, and ``":"``.
+
+        Returns
+        -------
+        value : bool, int, float, or numpy.ndarray
+            Whether the option was set or its value.
+
+            If the option was not set, return ``False``. Otherwise,
+            the return value depends on the choice of the option.
+
+            - options ``"B"``, ``"J"``, ``"U"``, ``"g"``, ``"n"``, ``"p"``,
+              and ``"s"``: return ``True``
+            - ``"I"``: 2-element array for the increments (float)
+            - ``"R"``: 4-element array for the region (float)
+            - ``"V"``: the verbose level (int)
+            - ``"X"``: the xshift (float)
+            - ``"Y"``: the yshift (float)
+            - ``"a"``: geometry of the dataset (int)
+            - ``"b"``: the option was set for input or output (int)
+            - ``"f"``: the option was set for input or output (int)
```

Comment:
Actually, are b and f the same? binary (b) and coltypes (f)?
```suggestion
            - ``"b"``: the option that was set for input or output (int)
            - ``"f"``: the option that was set for input or output (int)
```

Reply:
No, they're not the same.

Initially, I thought the function `GMT_Get_Common` should return the value of the option. For example, after running a command like `gmt plot input.txt -R0/10/0/10 -JX10c/10c -Baf -X2c -Y2c -bi1f -fi0x,1y`, `GMT_Get_Common` should return `X10c/10c` for `J` option and return `i0x,1y` for `f` option, but it doesn't. 

That's because the `GMT_Get_Common` function can only return a status (the function return value) or a double array (function parameter passed by reference). Thus, for options like `-b` and `-f`, the function can only tell if the option was set for input or output (i.e., if `-bi`/`-fi` or `-bo`/`-fo` was specified).

Reply:
Ah ok, thanks for the explanation. I think your changes in https://github.com/GenericMappingTools/pygmt/pull/2500#discussion_r1173233018 look better.